weld/ast.rs

```diff
@@ -75,11 +75,136 @@ impl fmt::Display for ScalarKind {
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum BuilderImplementationKind {
+    Local,
+    Global
+}
+
+impl fmt::Display for BuilderImplementationKind {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use ast::BuilderImplementationKind::*;
+        let text = match *self {
+            Local => "local",
+            Global => "global",
+        };
+        f.write_str(text)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub struct Annotations {
+    builder_implementation: Option<BuilderImplementationKind>,
+
+    predicate: Option<bool>,
+    vectorize: Option<bool>,
+    tile_size: Option<i32>,
+    grain_size: Option<i32>,
+
+    size: Option<i32>,
+    branch_selectivity: Option<i32>, // Fractions of 10,000.
+    num_keys: Option<i32>,
+}
+
+impl Annotations {
+    pub fn new() -> Annotations {
+        return Annotations {
+            builder_implementation: None,
+            predicate: None,
+            vectorize: None,
+            tile_size: None,
+            grain_size: None,
+            size: None,
+            branch_selectivity: None,
+            num_keys: None,
+        }
+    }
+
+    pub fn get_builder_implementation(&self) -> Option<BuilderImplementationKind> {
+        self.builder_implementation.clone()
+    }
+
+    pub fn set_builder_implementation(&mut self,
+                                      builder_implementation: &str) {
```

Comment:
Would be better if this takes a BuilderImplementationKind instead of a string; why do the parsing here instead of in the parser?